traffic_ops/app/t/api/1.2/user_admin.t

```diff
@@ -0,0 +1,126 @@
+package main;
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+use Mojo::Base -strict;
+use Test::More;
+use Test::Mojo;
+use DBI;
+use Data::Dumper;
+use strict;
+use warnings;
+use Schema;
+use Test::TestHelper;
+use Fixtures::TmUser;
+use Fixtures::Deliveryservice;
+use Digest::SHA1 qw(sha1_hex);
+use Data::Dumper;
+
+#no_transactions=>1 ==> keep fixtures after every execution, beware of duplicate data!
+#no_transactions=>0 ==> delete fixtures after every execution
+
+BEGIN { $ENV{MOJO_MODE} = "test" }
+
+sub run_ut {
+	my $t = shift;
+	my $schema = shift;
+	my $login_user = shift;
+	my $login_password = shift;
+	
+	Test::TestHelper->unload_core_data($schema);
+	Test::TestHelper->teardown( $schema, 'Log' );
+	Test::TestHelper->teardown( $schema, 'Role' );
+	Test::TestHelper->teardown( $schema, 'TmUser' );
+
+	Test::TestHelper->load_core_data($schema);
+
+	my $tenant_id = $schema->resultset('TmUser')->find( { username => $login_user } )->get_column('tenant_id');
+	my $tenant_name = defined ($tenant_id) ? $schema->resultset('Tenant')->find( { id => $tenant_id } )->get_column('name') : "null";
+
+	# Verify the user
+	ok my $user = $schema->resultset('TmUser')->find( { username => $login_user } ), 'Does the portal user exist?';
+	
+	ok $t->post_ok( '/login', => form => { u => $login_user, p => $login_password} )->status_is(302);
+		
+	#adding a user
+	my $addedUserName = "user1";
+	my $addedUserEmail = "abc\@z.com";
+
+	ok $t->post_ok('/api/1.2/users' => {Accept => 'application/json'} => json => {
+        	"username" => $addedUserName, "fullName"=>"full name", "email" => $addedUserEmail, "localPassword" => "pass", "confirmLocalPassword"=> "pass", "role" => 4 })
+        	->status_is(200)->or( sub { diag $t->tx->res->content->asset->{content}; } )
+		->json_is( "/response/username" =>  $addedUserName )
+		->json_is( "/response/email" =>  $addedUserEmail)
+		->json_is( "/response/tenantId" =>  $tenant_id) #tenant Id not set - getting the tenant id from the user
+        	    , 'Failed adding user?';
```

Comment:
this test message doesn't seem right either. Maybe "Successfully added user?" would be better.